src/uu/sum/src/sum.rs

```diff
@@ -63,6 +69,7 @@ fn sysv_sum(mut reader: Box<dyn Read>) -> (usize, u16) {
     ret = (ret & 0xffff) + (ret >> 16);
     ret = (ret & 0xffff) + (ret >> 16);
 
+    let blocks_read = div_ceil(bytes_read, 512);
```

Comment:
please a comment explaining the why here, thanks

Reply:
Done.

Reply:
Actually, the original version is better even for non-power-of-two values of `b` as well, though not as substantially: https://godbolt.org/z/njP8qvqs8